marshmallow_recipe/fields.py

```diff
@@ -757,11 +757,14 @@ def _deserialize(self, value: Any, attr: Any, data: Any, **kwargs: Any) -> Any:
                 raise m.ValidationError(self.default_error.format(input=value, choices=self.choices))
 
         @staticmethod
-        def _validate_enum(enum_type: Any) -> None:
+        def _extract_enum_value_type(enum_type: Any) -> type[str | int]:
             if not issubclass(enum_type, enum.Enum):
                 raise ValueError(f"Enum type {enum_type} should be subtype of Enum")
-            if not issubclass(enum_type, str):
-                raise ValueError(f"Enum type {enum_type} should be subtype of str")
+            if issubclass(enum_type, str):
+                return str
+            if issubclass(enum_type, int):
+                return int
```

Comment:
It would've been nicer to extract the type instead of doing ad-hoc checks but according to the implementation of `StrEnum`/`IntEnum` there's not easy way to do it, right? 😅

Reply:
hmm, enum with strings and integers can be declared without using StrEnum/IntEnum and there is an example in tests.

I used the same approach that was before my changes but extended it to support int in addition to str.

I also checked [native enum support which was added in marshmallow 3.18](https://github.com/marshmallow-code/marshmallow/blob/31d67b9c759aa1f3b307c1df8c0dfaa97e733661/src/marshmallow/fields.py#L1864), I had thoughts to use it or reuse implementation for v2 but than decided not to change approach used in marshmallow-recipe

Reply:
> enum with strings and integers can be declared without using StrEnum/IntEnum and there is an example in tests.

Exactly, it's just a mix of `Enum` and `int`/`str`, my idea was to get superclasses of resulting enum, filter out enum-related classes and what's left should be a single _alien_ class which is `int` or `str` in this case but could also be anything else

Reply:
what are the benefits of this approach?